sim/battle.ts

```diff
@@ -1509,7 +1509,9 @@ export class Battle {
 			trappedBySide.push(sideTrapped);
 			stalenessBySide.push(sideStaleness);
 			side.faintedLastTurn = side.faintedThisTurn;
+			side.pokemonFaintedLastTurn = side.pokemonFaintedThisTurn;
 			side.faintedThisTurn = false;
+			side.pokemonFaintedThisTurn = null;
```

Comment:
Do we need all of these? Can `faintedThisTurn` and `pokemonFaintedThisTurn` be merged into one property?